docs/src/test-api/class-testoptions.md

```diff
@@ -89,42 +178,195 @@ When connect options are specified, default [`property: Fixtures.browser`], [`pr
+
+export default defineConfig({
+  use: {
+    contextOptions: {
+      viewport: viewport: { width: 1280, height: 720 },
+    },
+  },
+});
+```
+
 ## property: TestOptions.deviceScaleFactor = %%-context-option-devicescalefactor-%%
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    deviceScaleFactor: 2
+  },
+});
+```
+
 ## property: TestOptions.extraHTTPHeaders = %%-context-option-extrahttpheaders-%%
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    extraHTTPHeaders: {
+      'X-My-Header': 'value',
+    },
+  },
+});
+```
+
 ## property: TestOptions.geolocation = %%-context-option-geolocation-%%
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    geolocation: { longitude: 12.492507, latitude: 41.889938 },
+  },
+});
+```
+
 ## property: TestOptions.hasTouch = %%-context-option-hastouch-%%
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    hasTouch: true
+  },
+});
+```
+
 ## property: TestOptions.headless = %%-browser-option-headless-%%
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    headless: false
+  },
+});
+```
+
 ## property: TestOptions.httpCredentials = %%-context-option-httpcredentials-%%
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    httpCredentials: {
+      username: 'user',
+      password: 'pass',
+    },
+  },
+});
+```
+
 ## property: TestOptions.ignoreHTTPSErrors = %%-context-option-ignorehttpserrors-%%
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    ignoreHTTPSErrors: true,
+  },
+});
+```
+
 ## property: TestOptions.isMobile = %%-context-option-ismobile-%%
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    isMobile: false,
+  },
+});
+```
+
 ## property: TestOptions.javaScriptEnabled = %%-context-option-javascriptenabled-%%
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    javaScriptEnabled: false,
+  },
+});
+```
+
 ## property: TestOptions.launchOptions
 * since: v1.10
 - type: <[Object]>
 
 Options used to launch the browser, as passed to [`method: BrowserType.launch`]. Specific options [`property: TestOptions.headless`] and [`property: TestOptions.channel`] take priority over this.
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    launchOptions: {
```

Comment:
Put this to a "chromium" project.

Reply:
would --load-extension work on a normal launch()? I thought this works only with launchPersistentContext.

Reply:
Good point. We can use `--start-maximized` as an example.